### PR TITLE
Add customizable Header content field

### PR DESCRIPTION
**Pull request by wes-otf**
This is a starting point for #3427. 

With this functionality the wagtail admin panel can be used to add a custom HTML header - very similarly to the footer. **Unlike the footer**, this field is not required and when left blank the default header will be used. This goal is to use this to be able to recreate the WordPress site navbar/header.

## Screenshots
**Using a blank "Header content" wagtail field:**

_The empty "Header Content" Wagtail admin field_
Screenshot 2023-10-06 at 1 32 48 PM

_The usual nav bar_
Screenshot 2023-10-06 at 2 23 19 PM

**Using a custom HTML header (rough-cut, not finalized)**

_The start of the custom HTML "Header content" field_
Screenshot 2023-10-06 at 2 15 05 PM

_The custom header that shows resemblance to the WordPress header (needs more tweaking)_ 
Screenshot 2023-10-06 at 2 14 38 PM

Being that this is my first PR feel free to give any critique you see necessary! Anything is helpful. Cheers!

### CODE REVIEW

File: hypha/templates/base.html [hunk 1]
```diff
@@ -72,109 +72,113 @@
...
-                            <svg class="header__icon header__icon--open-search header__icon--open-search-menu-closed icon icon--mobile-menu"><use xlink:href="#magnifying-glass"></use></svg>
-                            <svg class="header__icon header__icon--close-search header__icon--close-search-menu-closed icon icon--mobile-menu"><use xlink:href="#cross"></use></svg>
-                        </button>
-                        <button class="button button--left-space js-mobile-menu-toggle" aria-haspopup="true">
-                            <svg class="icon icon--mobile-menu"><use xlink:href="#mobile-menu-toggle"></use></svg>
-                        </button>
-                    </div>
-
-                    <section class="header__menus header__menus--desktop">
-                        {% cache 3600 navigation__primary current_site %}
```
Comment (frjo) [inline hypha/templates/base.html:115]: It is only this primary nav that should be replaced. Logo etc. should always be in place.

File: hypha/public/utils/models.py [hunk 2]
```diff
@@ -320,6 +320,12 @@ class Meta:
         ),
     )
 
+    header_content = models.TextField(
```
Comment (frjo) [inline hypha/public/utils/models.py]: Maybe rename this to "Front page navigation".

File: hypha/public/utils/models.py [hunk 3]
```diff
@@ -320,6 +320,12 @@ class Meta:
...
+        "Header content",
+        help_text=_("This will overwrite the default header, html tags is allowed."),
```
Comment (frjo) [inline hypha/public/utils/models.py]: This is good, let the default stay until a custom nav is added.

When we finally scrap the public side of Hypha the default will likely by "My dashboard etc.", as it is on all the apply pages.

**Review by frjo — CHANGES_REQUESTED** [review]

File: hypha/templates/base-apply.html [hunk 4]
```diff
@@ -134,7 +143,7 @@
                 </section>
 
                 <div class="header__button-container">
-                    {% if latest_notifications %}
+                    {% if latest_notifications %}.
```
Comment (frjo) [inline hypha/templates/base-apply.html]: A mistake period it look like?

File: hypha/templates/base-apply.html [hunk 5]
```diff
@@ -102,7 +102,12 @@
                 </div>
 
                 <section class="header__menus header__menus--desktop">
-                    {% include "navigation/primarynav-apply.html" %}
+                    {% if settings.utils.SystemMessagesSettings.nav_content|safe and not request.user.is_authenticated %}
```
Comment (frjo) [inline hypha/templates/base-apply.html]: I think we can skip the "safe" filter in the if statement.

Comment (frjo) [reply hypha/templates/base-apply.html]: For this intermediate version we can skip the "and not request.user.is_authenticated" part as well.

OTF will need this when they launch the new public site. When that happens the "primarynav*.html" will be obsolete.

We need to work on this when we completely remove the public site in any case.

**Review by frjo — CHANGES_REQUESTED** [review]

**Comment by frjo** [comment]
@wes-otf Works well but needs some minor fixes.

File: hypha/templates/base-apply.html (hunk 5 above)
Comment (wes-otf) [reply hypha/templates/base-apply.html]: Ugh - result of a sloppy copy & paste. Should be all set now.

**Review by frjo — APPROVED** [review]

**Pull request merged**